server/static/page.jsx

```diff
@@ -1,8 +1,404 @@
-let model = {};
+let model = {
+  lastError: null,
+  lookingUpShot: {},
+  lookingUpDevice: {},
+};
 
 class Page extends React.Component {
   render() {
-    return <span>I am a page</span>;
+    return <div>
+      { this.props.lastError ? <ErrorMessage error={this.props.lastError} /> : null }
+      <LookUpShot {...this.props.lookingUpShot} />
+      <LookUpDevice {...this.props.lookingUpDevice} />
+      <BlockDevice {...this.props} />
+      <BlockShot {...this.props} />
+    </div>;
+  }
+}
+
+class ErrorMessage extends React.Component {
+  render() {
+    let description = this.props.error.description;
+    let body = this.props.error.body;
+    let props = Object.assign({}, this.props.error);
+    delete props.description;
+    delete props.body;
+    return <div>
+      <h1>Error</h1>
+      <div>{description}</div>
+      <pre>{JSON.stringify(props, null, "  ")}</pre>
+      <pre style={{wordBreak: "normal"}}>{body}</pre>
+    </div>;
+  }
+}
+
+class LookUpShot extends React.Component {
+  render() {
+    return <fieldset>
+      <legend>Look up shot</legend>
+
+      <div>
+        <form onSubmit={this.onSubmit.bind(this)} id="lookup-shot-form">
+          Look up shot:
+          <input
+          placeholder="shot ID, shot URL, or image URL"
+          defaultValue={this.props.searchTerm}
+          ref={input => this.input = input}
+          id="lookup-shot-input"
+          type="text" />
+          <button type="submit">Look up</button>
+        </form>
+      </div>
+      <hr />
+      { this.props.notFound ? <div>No shot found with term {this.props.searchTerm}</div> : null }
+      { this.props.shotInformation ? <ShotInformation shot={this.props.shotInformation} /> : null }
+      { this.props.shotSearching ? "Searching..." : null }
+    </fieldset>;
+  }
+
+  async onSubmit(event) {
+    event.preventDefault();
+    let searchTerm = this.input.value;
+    model.lookingUpShot.notFound = false;
+    model.lookingUpShot.shotInformation = null;
+    model.lookingUpShot.shotSearching = true;
+    render();
+    let resp = await fetch(`/lookup-shot?term=${encodeURIComponent(searchTerm)}`);
+    model.lookingUpShot.shotSearching = false;
+    if (resp.status === 404) {
+      model.lookingUpShot.notFound = true;
+      render();
+      return;
+    }
+    if (! resp.ok) {
```

Comment:
Not sure it really matters, but should `model.lookingUpShot.notFound` be `true` in this case, too?